.github/STYLE_GUIDE.md

```diff
@@ -0,0 +1,344 @@
+
+### Module Docstring
+- Include a module-level docstring at the beginning of each Python file.
+- Use triple double-quotes for multi-line docstrings.
+
+    Example:
+
+    ```python
+    """Module-level docstring.
+
+    This module provides utility functions for handling calculations.
+    """
+
+    # Rest of the module code
+    ```
+
+### Class Docstring
+- Include a class-level docstring immediately below the class definition.
+- Briefly describe the purpose and usage of the class.
+
+    Example:
+
+    ```python
+    class Calculator:
+        """A simple calculator class.
+
+        This class provides basic arithmetic operations such as addition and subtraction.
+        """
+
+        def __init__(self):
+            # Constructor implementation
+    ```
+
+### Function Docstring
+- Include a function-level docstring immediately below the function definition.
+- Provide a clear description of the function's purpose, parameters, and return values.
+
+    Example:
+
+    ```python
+    def calculate_sum(numbers):
+        """Calculate the sum of a list of numbers.
+
+        Args:
+            numbers (list): A list of numerical values.
+
+        Returns:
+            float: The sum of the input numbers.
+        """
+        # Function implementation
+
+    ```
+
+<br/>
+
+## Typing
+Python's optional type hints, introduced in PEP 484 and expanded in subsequent PEPs, provide a way to statically indicate the type of variables and function parameters. Proper use of typing can enhance code readability, maintainability, and catch certain types of errors early in the development process.
+
+### General Guidelines
+1. **Use Type Hints:**
+
+    Type hints should be used consistently to indicate the expected types of variables and function parameters.
+
+    Example:
+    
+    ```python
+    def add_numbers(a: int, b: int) -> int:
+        return a + b
+    ```
+
+2. **Avoid Redundant Type Hints:**
+
+    Avoid providing type hints when the type is obvious from the variable name or the context.
+
+    Example:
+    
+    ```python
+    # Good
+    name: str = "John"
+
+    # Avoid unnecessary type hint
+    age = 30  # Type is clear without specifying it
+    ```
+
+3. **Use Expressive Variable Names:**
+
+    Choose variable names that convey meaning and make type hints redundant.
+
+    Example:
+
+    ```python
+    def calculate_area(length: float, width: float) -> float:
+        return length * width
+    ```
+
+4. **Be Consistent with Typing Styles:**
+
+    Choose a consistent style for type hints, either using the ```:``` notation or the ```->``` notation for function return types.
+
+    Example:
+
+    ```python
+    # Consistent style with `:`
+    def greet(name: str) -> None:
+        print(f"Hello, {name}!")
+
+    # Consistent style with `->`
+    def multiply(a: int, b: int) -> int:
+        return a * b
+    ```
```

Comment:
I don't understand what this is saying. Both examples use `->`?

Reply:
well damn, that's what I get for trusting chatgpt